js/gdm/loginDialog.js

```diff
@@ -725,26 +841,40 @@ const LoginDialog = new Lang.Class({
 
     _reset: function() {
         this._userVerifier.clear();
+        this._userVerifier.resetFailCounter();
 
         this._updateSensitivity(true);
         this._promptMessage.hide();
         this._user = null;
         this._passwordHintButton.visible = false;
+        this._passwordResetButton.visible = false;
+        this._passwordResetCode = null;
         this._verifyingUser = false;
 
         if (this._disableUserList)
             this._hideUserListAndLogIn();
         else
             this._showUserList();
+
+        if (this._holdForAnswer) {
+            this._holdForAnswer.release();
+            this._holdForAnswer = null;
+        }
     },
 
     _verificationFailed: function() {
+        // Nothing to do if we were just reset after too many failed verifications.
```

Comment:
Nope, because they were already just cleared by _reset(). It might be more clear to write if (!this._verifyingUser) instead, but checking this._user seems worthwhile since that's the variable that's actually being used.